src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Message.java

```diff
@@ -16,6 +17,11 @@
     private final HealthAuthorityNameAndAddress healthAuthorityNameAndAddress =
         HealthAuthorityNameAndAddress.fromString(extractSegment(HealthAuthorityNameAndAddress.KEY_QUALIFIER));
 
+    @Getter(lazy = true)
+    private final Optional<RequesterNameAndAddress> requesterNameAndAddress =
+        extractOptionalSegment(RequesterNameAndAddress.KEY_QUALIFIER)
+            .map(RequesterNameAndAddress::fromString);
```

Comment:
Ported this style from NHAIS and I really don't like it very much; having an `Optional` field is like fingernails down a chalkboard to me. But I understand it in conjunction with the lazy getter given by lombok.

Question then is: is the laziness really needed? Do we expect to be processing so much data that performance at that scale is important?

Reply:
It's that or null. There was some reasoning behind choosing laziness that I can't recall right now. It's either this or parsing everything in constructor. I think I preferred to defer parsing everything at constructor level but if you think code would look better without it then I have no objections

Reply:
I see three options:
**Option 1: NHAIS style**
\+ Consistent with sibling NHAIS project.
\+ Utilises lombok to reduce boilerplate.
\+ The scan of segments to find the data happens at most once.
– `Optional` as fields is discouraged - either need to suppress warnings or live with warnings
```
    @Getter(lazy = true)
    private final Optional<RequesterNameAndAddress> requesterNameAndAddress =
        extractOptionalSegment(RequesterNameAndAddress.KEY_QUALIFIER)
            .map(RequesterNameAndAddress::fromString);
```

**Option 2: Fieldless**
\+ Intended use of `Optional`
\+ Familiar java style
– Scans the segments for the data every time the method is called
```
    public Optional<RequesterNameAndAddress> getRequesterNameAndAddress() {
        return extractOptionalSegment(RequesterNameAndAddress.KEY_QUALIFIER)
            .map(RequesterNameAndAddress::fromString);
    }
```
**Option 3: `Lazy` class**
\+ Segments are scanned at most once
\+ Minimal boilerplate, without lombok
– More complex generic type
```
@RequiredArgsConstructor
public class Lazy<T> {
    private T value;
    private boolean evaluated;
    private final Supplier<T> supplier;

    public T get() {
        if (!evaluated) {
            value = supplier.get();
            evaluated = true;
        }
        return value;
    }
}
```
```
    private final Lazy<Optional<RequesterNameAndAddress>> requesterNameAndAddress = new Lazy<>(
        () -> extractOptionalSegment(RequesterNameAndAddress.KEY_QUALIFIER)
            .map(RequesterNameAndAddress::fromString)
    );
```

Reply:
Team agreed to stick with the NHAIS style